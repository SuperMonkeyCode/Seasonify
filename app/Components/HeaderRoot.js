import React from 'react';
import { Header, Title, Button, Left, Right, Body, Icon } from 'native-base';

export default function HeaderRoot({
  title,
  navigation,
}: {
  title: string,
  navigation: Object,
}) {
  const openDrawer = () => {
    navigation.navigate('DrawerOpen');
  };
  return (
    <Header>
      <Left>
        <Button transparent onPress={openDrawer}>
          <Icon name="menu" />
        </Button>
      </Left>
      <Body>
        <Title>{title}</Title>
      </Body>
      <Right />
    </Header>
  );
}
