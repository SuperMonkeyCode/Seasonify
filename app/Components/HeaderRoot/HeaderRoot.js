/* @flow */
import { Header, Title, Button, Left, Right, Body, Icon } from 'native-base';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  title: string,
  navigation: Object,
  /* connect */
  /* state */
  /* handlers */
  handleDrawer: Function,
};

const enhance = compose(
  pure,
  withHandlers({
    handleDrawer: ({ navigation }: Props) => () => {
      navigation.navigate('DrawerOpen');
    },
  }),
);

function HeaderRoot({ title, handleDrawer }: Props) {
  return (
    <Header>
      <Left>
        <Button transparent onPress={handleDrawer}>
          <Icon name="menu" />
        </Button>
      </Left>
      <Body>
        <Title>{title}</Title>
      </Body>
      <Right />
    </Header>
  );
}

export default enhance(HeaderRoot);
