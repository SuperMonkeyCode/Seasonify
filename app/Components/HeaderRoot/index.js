/* @flow */
import HeaderRoot from './HeaderRoot';

export default HeaderRoot;
