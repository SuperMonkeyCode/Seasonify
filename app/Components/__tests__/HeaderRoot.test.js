import React from 'react';
import HeaderRoot from '../HeaderRoot';

function setup(specificProps = {}) {
  const props = {
    title: 'Some tvshow',
    navigation: { navigate: jest.fn() },
    ...specificProps,
  };
  const component = shallow(<HeaderRoot {...props} />);
  return {
    component,
    props,
  };
}

describe('Rendering', () => {
  it('should render', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});

describe('Events', () => {
  it('should call navigation.navigate', () => {
    const { component, props } = setup();
    component.find('Styled(Button)').simulate('press');
    expect(props.navigation.navigate).toBeCalledWith('DrawerOpen');
  });
});
