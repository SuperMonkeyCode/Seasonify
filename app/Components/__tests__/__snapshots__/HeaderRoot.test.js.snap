// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should render 1`] = `
<Styled(Header)>
  <Styled(Left)>
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="menu"
      />
    </Styled(Button)>
  </Styled(Left)>
  <Styled(Body)>
    <Styled(Title)>
      Some tvshow
    </Styled(Title)>
  </Styled(Body)>
  <Styled(Right) />
</Styled(Header)>
`;
